common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/SignalizedIntersection.h

```diff
@@ -0,0 +1,138 @@
+   *
+   * @return list of const lanelets
+   */
+  ConstLanelets getExitLanelets() const;
+
+  /**
+   * @brief Returns the carma traffic signals in the specified entry lanelet in the intersection
+   * @param entry lanelet in the intersection to get the traffic signals for
+   *
+   * @return vector of const traffic signals. Empty list if the lanelet is not an entry lanelet or not in the intersection
+   */
+  std::vector<CarmaTrafficLightConstPtr> getTrafficSignals(const ConstLanelet& llt) const;
+
+  /**
+   * @brief Returns the stop line of the specified lanelet in the intersection
+   *
+   * @return gets a const stop line for a lanelet, if there is one
+   */
+  Optional<lanelet::ConstLineString3d> getStopLine(const ConstLanelet& llt) const;
+
+  /**
+   * @brief Adds the lanelet into the specified section of the intersection
+   * @param lanelet to add  
+   * @param section section of the intersection to add to (entry, exit, interior)
+   * 
+   * NOTE: If entry lanelet is being added, the user must make sure that the regulatory element refers the lanelet back
+   *       For maps, laneletMapPtr->update(lanelet, regem) for full connection
+   */
+  void addLanelet(const Lanelet& lanelet, IntersectionSection section);
+
+  /**
+   * @brief Removes the lanelet from the intersection
+   * @param lanelet to remove
+   * @return true if successful, false if lanelet was not found  
+   * 
+   * NOTE: If entry lanelet is being removed, the user must make sure that they remove the regulatory element from the lanelet
+   *       For maps, laneletMapPtr->remove(lanelet, regem) for removing all connection
+   */
+  bool removeLanelet(const Lanelet& lanelet);
+
+  /**
+   * @brief Static helper function that creates a minimum gap  based on the provided double, start, end lines, and the
+   * affected participants
+   *
+   * @param id The lanelet::Id of this object
+   * @param entry Entry lanelets of the intersection (which this regem also refers)
+   * @param exit Exit lanelets of the intersection
+   * @param interior Interior lanelets of the intersection
+   *
+   * @return RegulatoryElementData containing all the necessary information to construct a minimum gap  element
+   */
+  static std::unique_ptr<lanelet::RegulatoryElementData> buildData(Id id, const Lanelets& entry, const Lanelets& exit, const Lanelets& interior);
+
+  /**
+   * @brief Constructor required for compatability with lanelet2 loading
+   *
+   * @param data The data to initialize this regulation with
+   */
+  explicit SignalizedIntersection(const lanelet::RegulatoryElementDataPtr& data);
+
+protected:
+  // the following lines are required so that lanelet2 can create this object when loading a map with this regulatory
+  // element
+  friend class RegisterRegulatoryElement<SignalizedIntersection>;
+  SignalizedIntersection(Id id, const AttributeMap& attributes, const Lanelets& entry, const Lanelets& exit, const Lanelets& interior);
```

Comment:
Is this constructor with the attribute map needed? It's not part of the element registration process.

Reply:
Nope not needed. deleted it